extensions/ql-vscode/src/run-queries.ts

```diff
@@ -657,6 +675,7 @@ export async function compileAndRunQueryAgainstDatabase(
   progress: ProgressCallback,
   token: CancellationToken,
   templates?: messages.TemplateDefinitions,
+  queryInfo?: LocalQueryInfo,
```

Comment:
So, this value will only be available for queries directly initiated by the user? Contextual queries won't have the `queryInfo`? That's fine, but I just want to be sure I understand. Could you add a comment here to explain that?

Reply:
Yes, that's exactly right. There's a few places where queries aren't initiated by the user (as you say, contextual queries, and also test code). In these cases we don't have a sensible place to put the structured logs, nor any reason to store them, so we don't create a log for these cases.